### PR TITLE
i3s: support compressed textures

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: modules/i3s/src/helpers/i3s-nodepages-tiles.js [hunk 1]
```diff
@@ -102,16 +106,46 @@ export default class I3SNodePagesTiles {
...
+      this.textureDefinitionsSelectedFormats.push(selectedFormat);
+    }
+  }
+
+  _getSupportedTextureFormats() {
+    const result = [];
+    const supportedCompressedFormats = getSupportedGPUTextureFormats();
+    // List of possible in i3s formats:
+    // https://github.com/Esri/i3s-spec/blob/master/docs/1.7/textureSetDefinitionFormat.cmn.md
+    if (supportedCompressedFormats.has('etc2')) {
```
Comment (ibgreen) [inline modules/i3s/src/helpers/i3s-nodepages-tiles.js:141]: @Tamrat-B FYI

File: modules/i3s/src/lib/parsers/parse-i3s-tile-content.js [hunk 2]
```diff
@@ -27,7 +35,17 @@ export async function parseI3STileContent(arrayBuffer, tile, tileset, options) {
 
   if (tile.textureUrl) {
     const url = getUrlWithToken(tile.textureUrl, options.token);
-    tile.content.texture = await load(url, ImageLoader);
+    const loader = FORMAT_LOADER_MAP[tile.textureFormat] || ImageLoader;
+    tile.content.texture = await load(url, loader);
+    if (loader === CompressedTextureLoader) {
+      tile.content.texture = {
+        compressed: true,
+        mipmaps: false,
```
Comment (ibgreen) [inline modules/i3s/src/lib/parsers/parse-i3s-tile-content.js:43]: The compressed textures do not have mipmaps?

**Review by ibgreen — APPROVED** [review]

File: modules/i3s/src/lib/parsers/parse-i3s-tile-content.js (hunk 2 above)
Comment (belom88) [reply modules/i3s/src/lib/parsers/parse-i3s-tile-content.js:43]: This flag is used in "luma" to check if we have to run `this.gl.generateMipmap(this.target);`. For compressed texture `generateMipmap` cause error.

Comment (ibgreen) [reply modules/i3s/src/lib/parsers/parse-i3s-tile-content.js:43]: Ideally luma should not generate mipmaps if `compressed` is true. Maybe add a comment explaining that this is what it means.

**Pull request merged**